src/Illuminate/Database/Concerns/BuildsQueries.php

```diff
@@ -296,6 +296,17 @@ public function first($columns = ['*'])
         return $this->take(1)->get($columns)->first();
     }
 
+    /**
+     * Execute the query and get the first result in Array.
+     *
+     * @param  array|string  $columns
+     * @return \Illuminate\Database\Eloquent\Model|array|static|null
```

Comment:
Due to the `(array)` cast the function will always return an array. Model and static types are not possible here. Even `(array) null;` will result in an empty array.